spec/src/main/asciidoc/chapters/language/expressions.adoc

```diff
@@ -75,25 +75,57 @@ An _identifier expression_, with syntax given by `id_expression`, is assigned th
+| `length` | 1 | String | Integer | Evaluates to the length of string to which its argument evaluates.
+| `lower` | 1 | String | String | Evaluates to the lowercase form of the string to which its argument evaluates.
+| `upper` | 1 | String | String | Evaluates to the uppercase form of the string to which its argument evaluates.
+| `left` | 2 | String, integer | String | Evaluates to a prefix of the string to which its first argument evaluates. The length of the prefix is given by the integer value to which its second argument evaluates.
+| `right` | 2 | String, integer | String | Evaluates to a suffix of the string to which its first argument evaluates. The length of the suffix is given by the integer value to which its second argument evaluates.
+| `#concat#` | At least one | All strings | String | Evaluates to the concatenation of its arguments.
+| `#substring#` | 2 or 3 | String, integer, integer | String | Evaluates to a specified substring of the first argument. The second and third arguments specify the starting position and length of the substring. The third argument is optional. If it is not specified, the substring from the starting position to the end of the string is returned. The first character of the string is at position `1`.
+| `#trim#` | 1^*^ | String | String | Trims a specified character from its last argument. If the character to be trimmed is not specified, the space character is trimmed. The optional `#trim_character#` specifies the character to be trimmed. The optional `#trim_specification#` controls whether the character is trimmed from the start and/or end of the string. By default, the character is trimmed from both start and end.
+| `#replace#` | 3 | All strings | String | Evaluates to a new string formed by replacing every occurrence of the second argument string within the first argument string with the third argument string.
+| `#locate#` | 2 or 3 | String, string, integer | Integer | Evaluates to the position at which one string occurs within a second string, optionally ignoring any occurrences that begin before a specified character position in the second string. It returns the first character position within the second string (after the specified character position, if any) at which the first string occurs, as an integer, where the first character of the second string is denoted by 1. That is, the first argument is the string to be searched for; the second argument is the string to be searched in; the optional third argument is an integer representing the character position at which the search starts (by default, 1, the first character of the second string). If the first string does not occur within the second string, 0 is returned.
+| `#size#` | 1 | Collection | Integer | Evaluates to the number of elements of the collection produced by evaluating its argument.
```

Comment:
This was a somewhat difficult to read at first. I think switching to the style of language used by the `length` function is helpful,

```suggestion
| `#size#` | 1 | Collection | Integer | Evaluates to the number of elements in the collection to which its argument evaluates.
```